src/lib/io/sol-iio.c

```diff
@@ -329,11 +383,19 @@ check_trigger(struct sol_iio_device *device)
     }
 
     len = sol_util_read_file(path, "%ms", &name);
+
     if (len < 0) {
-        SOL_INF("No current trigger for iio:device%d. Creating a sysfs one.",
-            device->device_id);
-        if (!create_sysfs_trigger(device))
-            return false;
+        if (strstartswith(config->trigger_name, HRTIMER_TRIGGER_PREFIX)) {
```

Comment:
`config->trigger_name` may be NULL here, you need to check this before startwith.